src/test/ui/parser/inverted-parameters.stderr

```diff
@@ -0,0 +1,41 @@
+error: expected one of `:` or `@`, found `bar`
+  --> $DIR/inverted-parameters.rs:14:24
+   |
+LL |     fn foo(&self, &str bar) {}
+   |                   -----^^^
+   |                   |    |
+   |                   |    expected one of `:` or `@` here
+   |                   help: declare the type after the parameter binding: `bar: &str`
+
+error: expected one of `:` or `@`, found `quux`
+  --> $DIR/inverted-parameters.rs:20:10
+   |
+LL | fn baz(S quux, xyzzy: i32) {}
+   |        --^^^^
+   |        | |
+   |        | expected one of `:` or `@` here
+   |        help: declare the type after the parameter binding: `quux: S`
```

Comment:
I'm thinking it may make sense to not provide a suggestion but rather a `help` message, otherwise we could confuse people that just forgot about the colon a non-sensical suggestion. For example, given `fn baz(quuz S) {}` we would suggest incorrect code.

Reply:
Ah, good catch. That is unfortunate. Something that's nice about the suggestion is that it illustrates the syntax, which is hard to describe with words alone. Would a suggestion like `<identifier>: <type>` with `Applicability::HasPlaceholders` suffice?

Reply:
Yes, I think that's an appropriate way to express it.